pkg/controller/servicebinding/secret.go

```diff
@@ -52,31 +88,34 @@ func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1
 	existingSecret, err := s.get()
 	if err != nil {
 		if errors.IsNotFound(err) {
-			_, err := resourceClient.Create(u, metav1.CreateOptions{})
+			logger.Info("Secret not found. Creating secret")
+			secretNew, err := resourceClient.Create(u, metav1.CreateOptions{})
 			if err != nil {
 				logger.Error(err, "Error creating secret")
 				return nil, err
 			}
 			logger.Info("Secret created")
-			return u, nil
+			return secretNew, nil
 		}
 		return nil, err
 	}
+
 	existingSecretData, _, _ := unstructured.NestedMap(existingSecret.Object, "data")
 	payloadInterim := make(map[string]interface{})
-	for k, v := range payload {
+	for k, v := range s.data {
 		payloadInterim[k] = base64.StdEncoding.EncodeToString(v)
 	}
 	comparisonResult := nestedMapComparison(existingSecretData, payloadInterim)
```

Comment:
we do not need to do any comparison anymore - instead it just enough to generate content hashes and compare them.

Reply:
Done